File: extras/scripts/postwritehooks/sample.py
"""
Scripts that are not written in PHP can be used.

This is a sample script that writes some data to a file.
"""

import sys
import time
from ConfigParser import SafeConfigParser

# Grab the first parameter, which will be a string.
record_key = sys.argv[1]

# The second paramter will be a comma-separated list containing
# zero or more children record keys.
children_record_keys = sys.argv[2].split(',')

# The third parameter will be the path to the MIK .ini file.
parser = SafeConfigParser()
parser.read(sys.argv[3])

# Print the parameters to a log.
file = open("/tmp/sample.py.txt", "a")
file.write("Record key in sample.py: %s\n" % record_key)
# Convert the array into string so we can print it to the log.
children_string = ', '.join(children_record_keys)
file.write("Children record keys in sample.py: %s\n" % children_string)
# Access a value from the .ini file.
file.write("CONTENTdm web services URL from sample.py: %s\n" % parser.get('FETCHER', 'record_key'))

File: extras/scripts/postwritehooks/sample.php
<?php

/**
 * All post-write hook scripts get the record key as the first parameter, a comma-
 * separated list of children record keys as the second, and the path to the MIK .ini
 * file as the third.
 *
 * This is a sample script that writes some data to a file.
 */

$record_key = trim($argv[1]);
$children_record_keys_string = trim($argv[2]);
$children_record_keys = explode(',', $children_record_keys_string);
$config_path = trim($argv[3]);

$config = parse_ini_file($config_path, true);

// Write some data from the parameters to a file.
file_put_contents('/tmp/task1.txt', "Record key from task1.php: $record_key\n", FILE_APPEND);
file_put_contents('/tmp/task1.txt', "Children record key from task1.php: " . implode(',', $children_record_keys) . "\n", FILE_APPEND);
file_put_contents('/tmp/task1.txt', "Output directory from MIK config: " . $config['WRITER']['output_directory'] . "\n", FILE_APPEND);
file_put_contents('/tmp/task1.txt', "Sample post-write hook script has finished\n", FILE_APPEND);
